Decode/EventHeaderEnumerator.cs

```diff
@@ -1148,25 +1152,13 @@ public EventHeaderItemInfo GetItemInfo()
         public EventHeaderItemInfo GetItemInfo(ReadOnlySpan<byte> eventDataSpan)
         {
             Debug.Assert(m_eventData.Length == eventDataSpan.Length);
-
-            if (m_state <= EventHeaderEnumeratorState.BeforeFirstItem)
-            {
-                throw new InvalidOperationException(); // PRECONDITION
-            }
-
             return new EventHeaderItemInfo(
                 eventDataSpan,
                 m_stackTop.NameOffset,
                 m_stackTop.NameSize,
                 new PerfItemValue(
                     eventDataSpan.Slice(m_dataPosCooked, m_itemSizeCooked),
-                    new PerfItemType(
-                        m_byteReader,
-                        m_fieldType.Encoding,
-                        m_fieldType.Format,
-                        m_elementSize,
-                        m_state == EventHeaderEnumeratorState.Value ? (ushort)1 : m_stackTop.ArrayCount,
-                        m_fieldType.Tag)));
+                    this.GetItemMetadata()));
```

Comment:
Note: The logic for creating the ItemMetadata value is starting to become non-trivial, so call GetItemMetadata instead of duplicating the logic.